server/src/controllers/roomsController.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import express from 'express';
+import { CommunicationUserIdentifier } from '@azure/communication-common';
+import { CommunicationAccessToken, CommunicationIdentityClient, TokenScope } from '@azure/communication-identity';
+import { Room, RoomsClient, RoomParticipant, CreateRoomOptions } from '@azure/communication-rooms';
+import {
+  CreateTestAppointmentResponse,
+  JoinTestAppointmentResponse,
+  RoomParticipantRole,
+  RoomParticipant as TestAppointmentRoomParticipant
+} from '../models/roomModel';
+import { ERROR_NO_USER_FOUND_IN_ROOM } from '../constants';
+
+export const createRoom = (identityClient: CommunicationIdentityClient, roomsClient: RoomsClient) => async (
+  _req: express.Request,
+  res: express.Response,
+  next: express.NextFunction
+): Promise<any> => {
+  try {
+    const presenter = await identityClient.createUser();
+    const attendee = await identityClient.createUser();
+
+    // Options payload to create a room
+    const createRoomOptions: CreateRoomOptions = {
+      participants: [
+        {
+          id: presenter,
+          role: RoomParticipantRole.presenter
+        },
+        {
+          id: attendee,
+          role: RoomParticipantRole.attendee
+        }
+      ]
+    };
+
+    // Create a room with the request payload
+    const createdRoom: Room = await roomsClient.createRoom(createRoomOptions);
+
+    // Formulating participants
+    const participants = createdRoom.participants.map(
+      (participant: RoomParticipant): TestAppointmentRoomParticipant => ({
+        id: (participant.id as CommunicationUserIdentifier).communicationUserId as string,
+        role: participant.role as RoomParticipantRole
+      })
+    );
+
+    // Formulate response
+    const response: CreateTestAppointmentResponse = {
+      roomId: createdRoom.id,
+      participants: participants
+    };
+
+    return res.send(response);
+  } catch (error) {
+    return next(error);
+  }
+};
+
+export const joinRoom = (identityClient: CommunicationIdentityClient, roomsClient: RoomsClient) => async (
+  req: express.Request,
+  res: express.Response,
+  next: express.NextFunction
+): Promise<any> => {
+  try {
+    const { roomId, userId } = req.query;
+
+    // Retrieve participants list
+    const participantsList = await roomsClient.getParticipants(roomId as string);
+
+    // Check if the user is part of participants
+    const foundUserParticipant: RoomParticipant | undefined = participantsList.find(
+      (participant: RoomParticipant) => (participant.id as CommunicationUserIdentifier).communicationUserId === userId
+    );
+
+    if (foundUserParticipant) {
```

Comment:
Suggestion to inverse the if check so we exit early. It's just for more readable code but it's not really important. Up to you if you want to change it. For example:

```
if (!foundUserParticipant)
{
    return 404;
}

// Retrieve token
```

This way the larger block of code for retrieving token doesn't need to be indented another level as it is outside the if block